src/test/scala/nl.knaw.dans.easy.solr4files/AppSpec.scala

```diff
@@ -69,24 +72,112 @@ class AppSpec extends TestSupportFixture {
       feedback.toString shouldBe s"Bag ${ uuidCentaur }: 7 times FileSubmittedWithContent, 2 times FileSubmittedWithOnlyMetadata"
     }
   }
 
+  it should "report problems when easy-auth-info is down" in {
+    // this tests the chain of error handling starting at AuthInfo
+    // other FileItem problems are tested with "updateFiles"
+    initVault()
+    val app = new StubbedSolrApp()
+    app.expectsHttpAsString(Failure(HttpStatusException("", HttpResponse("", SC_SERVICE_UNAVAILABLE, Map("Status" -> IndexedSeq("")))))) once()
+
+    val result = app.update(storeName, uuidCentaur)
+    inside(result) {
+      case Failure(MixedResultsException(_, HttpStatusException(_, HttpResponse(_, SC_SERVICE_UNAVAILABLE, _)))) =>
+    }
+  }
+
   it should "not stumble on difficult file names" in {
     initVault()
     assume(canConnectToEasySchemas)
-    val result = new StubbedSolrApp().update(storeName, uuidAnonymized)
+    val app = new StubbedSolrApp()
+    app.expectsHttpAsString(anonymousAuthItem(uuidAnonymized, "YYY")) anyNumberOfTimes()
+
+    val result = app.update(storeName, uuidAnonymized)
     inside(result) { case Success(feedback) =>
       feedback.toString shouldBe s"Bag ${ uuidAnonymized }: 3 times FileSubmittedWithContent"
     }
   }
 
+  "updateFiles" should "retrieve checksum, files size nor DDM.solrLiterals because file is not accessible" in {
+    val app = new StubbedSolrApp()
+    app.expectsHttpAsString(Success(
+      s"""{
+         |  "itemId":"$uuid/xy.z",
+         |  "owner":"someone",
+         |  "dateAvailable":"1992-07-30",
+         |  "accessibleTo":"NONE",
+         |  "visibleTo":"NONE"
+         |}""".stripMargin)
+    ) once()
+    class MockedDDM extends DDM(<empty/>)
+    class MockedBag extends Bag("pdbs", uuid, mock[Vault])
+    val result = app.updateFiles(mock[MockedBag], mock[MockedDDM], <files><file filepath="xy.z"/></files>)
+    result shouldBe Success(BagSubmitted(uuid.toString, Seq.empty))
+  }
+
+  it should "report problems when easy-auth-info returns invalid content" in {
+    clearVault()
+    write(testDir.resolve(s"vault/stores/pdbs/bags/$uuid/metadata/dataset.xml").toFile, "<ddm/>")
+    write(testDir.resolve(s"vault/stores/pdbs/bags/$uuid/metadata/files.xml").toFile, "<files><file filepath='xy.z'/></files>")
+    val authInfoJson = """{ "visibleTo":"ANONYMOUS" }""".stripMargin
+    val app = new StubbedSolrApp()
+    app.expectsHttpAsString(Success(authInfoJson))
+
+    val result = app.update(storeName, uuid)
+    inside(result) {
+      case Failure(MixedResultsException(_, e)) => e.getMessage shouldBe
+        s"""parse error [class org.json4s.package$$MappingException: No usable value for itemId
+           |Did not find value which can be converted into java.lang.String] for: ${ authInfoJson.toOneLiner }""".stripMargin
+    }
+  }
+
+  it should "report an invalid bag: file-item without a path" in {
+    val app = new StubbedSolrApp()
+    class MockedDDM extends DDM(<empty/>)
```

Comment:
if you use these mocks more often in this file (let's say, at least 3x), move them to the top of this file. If you use them more often in other files, make them into a fixture